optuna/importance/_fanova/_evaluator.py

```diff
@@ -79,67 +94,45 @@ def evaluate(
                 "`target=lambda t: t.values[0]` for the first objective value."
             )
 
-        distributions = _get_distributions(study, params)
-        if len(distributions) == 0:
-            return OrderedDict()
+        distributions = _get_distributions(study, params=params)
+        if params is None:
+            params = list(distributions.keys())
+        assert params is not None
 
         # fANOVA does not support parameter distributions with a single value.
         # However, there is no reason to calculate parameter importance in such case anyway,
         # since it will always be 0 as the parameter is constant in the objective function.
-        zero_importances = {name: 0.0 for name, dist in distributions.items() if dist.single()}
-        distributions = {name: dist for name, dist in distributions.items() if not dist.single()}
-
-        trials = []
-        for trial in _filter_nonfinite(
-            study.get_trials(deepcopy=False, states=(TrialState.COMPLETE,)), target=target
-        ):
-            if any(name not in trial.params for name in distributions.keys()):
-                continue
-            trials.append(trial)
-
-        trans = _SearchSpaceTransform(distributions, transform_log=False, transform_step=False)
-
-        n_trials = len(trials)
-        trans_params = numpy.empty((n_trials, trans.bounds.shape[0]), dtype=numpy.float64)
-        trans_values = numpy.empty(n_trials, dtype=numpy.float64)
+        non_single_distributions, single_distributions = _split_nonsingle_and_single_distributions(
+            distributions
+        )
```

Comment:
For me, previous implementations looks simpler than your changes.`_split_nonsingle_and_single_distributions` function is not informative very much. What do you think?

```suggestion
        single_distributions = {name: 0.0 for name, dist in distributions.items() if dist.single()}
        non_single_distributions = {name: dist for name, dist in distributions.items() if not dist.single()}
```

Reply:
It looks that this suggestion is still not applied :eyes: